clippy_lints/src/loops.rs

```diff
@@ -972,15 +984,41 @@ fn detect_manual_memcpy<'a, 'tcx>(
                     }
                 }
 
+                let mut end_expr = end;
+                let mut reduced_end_expr = false;
+
+                if_chain! {
+                    if let ExprKind::Binary(ref op, ref left, ref right) = end.node;
+                    if let BinOpKind::Add = op.node;
+                    then {
+                        if let Some(left) = sugg::Sugg::hir_opt(cx, left) {
+                            if left.to_string() == offset.value {
```

Comment:
I'm not too happy about the `to_string()` + comparison.

The other changes in this PR are great, but I'd prefer to postpone this entire `if_chain` until we have a proper expression minifier/optimizer

Reply:
I removed it, but this makes the `0..src.len()` check below useless until the expression minifier is implemented. Should I remove that as well?

Reply:
Oh... I guess, no need to keep around currently dead code